docs/pages/build-reference/app-versions.md

```diff
@@ -0,0 +1,97 @@
+
+```json
+{
+  "cli": {
+    "appVersionSource": "local"
+  },
+  "build": {
+    "staging": {
+      "distribution": "internal",
+      "android": {
+        "buildType": "apk"
+      }
+    },
+    "production": {
+      "autoIncrement": true
+    }
+  }
+}
+```
+
+</Collapsible>
+
+## Remote version source
+
+You can configure your project to rely on EAS servers to store and manage the version of your app, to do that add `{ "cli": { "appVersionSource": "remote" } }` in your **eas.json**. The remote version will be initialized with the value from a local project, but you can override it. If you want to set it to something else, or EAS CLI is not able to detect what version an app is on, you can configure it with `eas build:version:set` command. EAS is storing version information scoped by account, slug, platform, and application ID/bundle identifier, so for example, if you are building variants with different application IDs or bundle identifiers, versioning will be independent for each of them.
+
+If you want to build your project in Android Studio or Xcode, you can update your local project with the remote versions using `eas build:version:sync`.
+
+Enabling `autoIncrement` option in this mode is only possible for `versionCode`/`buildNumber`, make sure to remove those values from **app.json** otherwise, they will still show up in the manifest exposed by `expo-constants` even though the app itself will be on a correct version.
+
+### Limitations
+
+- `eas build:version:sync` command on Android does not support bare projects with multiple flavors, but the rest of the remote versioning functionality should work with all projects.
+- `autoIncrement` does not support the `version` option.
+- It's not supported if you are using EAS Update and runtime policy set to `"runtimeVersion": { "policy": "nativeVersion" }`.
+
+### Recommended workflow
+
+The main goal of this feature is to avoid manual changes to the project every time you are uploading a new archive to run it on a TestFlight or Play Store testing channels, but when you are doing a production release, the change should be explicit. We recommend always updating `version` field after a new build goes live in the store, especially if you are using `expo-updates` with an automatic runtime version policy.
+
+## Local version source
+
+By default all projects keep versions locally, EAS itself does not interact with it and just builds projects as they are, unless `autoIncrement` option is enabled. In the case of bare React Native projects, values in native code take priority, but `expo-constants` and `expo-updates` read values from **app.json**. If you rely on version values from a manifest, you should keep them in sync with native code. It's especially important if you are using EAS Update with runtime policy set to `"runtimeVersion": { "policy": "nativeVersion" }`, because it might result in the delivery of updates to the wrong version of an application.
```

Comment:
```suggestion
By default, the source of truth for a project versions is the local project source code itself. In this case, EAS does not write to the project, it reads the values and builds projects as they are.

You may opt in to auto incrementing versions locally with the `autoIncrement` option on a build profile, but it comes with some limitations.

In the case of bare React Native projects, values in native code take priority, and `expo-constants` and `expo-updates` read values from **app.json**. If you rely on version values from a manifest, you should keep them in sync with native code. Keeping these values in sync is especially important if you are using EAS Update with the runtime policy set to `"runtimeVersion": { "policy": "nativeVersion" }`, because it mismatched versions may result in the delivery of updates to the wrong version of an application. We recommend using `expo-application` to read the version instead of depending on values from **app.json**.
```